templates/demo-store/app/lib/session.server.ts

```diff
@@ -10,12 +10,12 @@ import {
  * swap out the cookie-based implementation with something else!
  */
 export class HydrogenSession {
-  constructor(
-    private sessionStorage: SessionStorage,
-    private session: Session,
-  ) {
-    this.session = session;
-    this.sessionStorage = sessionStorage;
+  #sessionStorage;
+  #session;
+
+  constructor(sessionStorage: SessionStorage, session: Session) {
+    this.#sessionStorage = sessionStorage;
+    this.#session = session;
```

Comment:
This was working fine but TS [parameter properties](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties) is a bit obscure. I think using assignments is easier to understand.
Plus, this can now rely on the private members of native JS (`this.#...`).